pkg/e2e/operators/operators.go

```diff
@@ -240,9 +240,22 @@ func checkUpgrade(h *helper.H, subNamespace string, subName string, packageName
 		ginkgo.It("should upgrade from the replaced version", func() {
 
 			// Get the CSV we're currently installed with
-			sub, err := h.Operator().OperatorsV1alpha1().Subscriptions(subNamespace).Get(context.TODO(), subName, metav1.GetOptions{})
-			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("failed trying to get Subscription %s in %s namespace", subName, subNamespace))
-			latestCSV := sub.Status.CurrentCSV
+			var latestCSV string
+			var sub *operatorv1.Subscription
+			var err error
+
+			pollErr := wait.PollImmediate(5*time.Second, 5*time.Minute, func() (bool, error) {
+				sub, err = h.Operator().OperatorsV1alpha1().Subscriptions(subNamespace).Get(context.TODO(), subName, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				latestCSV = sub.Status.CurrentCSV
+				if latestCSV != "" {
+					return true, nil
+				}
+				return false, nil
+			})
+			Expect(pollErr).NotTo(HaveOccurred(), fmt.Sprintf("failed trying to get Subscription %s in %s namespace: %s", subName, subNamespace, err.Error()))
```

Comment:
This will try to print a nil pointer `err`

Reply:
Just re-iterating this (good pick-up @dofinn), because it got lost in the sea of test errors. However, even if it's changed to `pollErr` it will still panic because on upon a successful exit, `pollErr` will be `nil`, the `Sprintf` gets called regardless, and accessing the `Error()` function will panic.

```suggestion
			Expect(pollErr).NotTo(HaveOccurred(), fmt.Sprintf("failed trying to get Subscription %s in %s namespace: %v", subName, subNamespace, err))
```

should fix it up.